### PR TITLE
Move IComponentFactory from EntryPoints namespace

### LINKED ISSUE #861: Move IComponentFactory from EntryPoints namespace
The current `IComponentFactory` types are in the `Microsoft.ML.Runtime.EntryPoints` namespace.

https://github.com/dotnet/machinelearning/blob/841ba785c58eb84fa0ce07de324f25d6e3dcbf94/src/Microsoft.ML.Core/EntryPoints/ComponentFactory.cs#L7-L12

We should move them to a more appropriate namespace, since `EntryPoint` doesn't really make sense, they are used outside of `EntryPoints`. I think they should go in the same namespace as the `ComponentCatalog`.

Also, we should remove the `IArgsComponent` interface, as I don't see it serving any purpose.

/cc @TomFinley @Zruty0 @codemzs

### PR DESCRIPTION
**Pull request by eerhardt**
I chose the same namespace as `ComponentCatalog` is in, which is the root `Microsoft.ML.Runtime` namespace. If we ever move `ComponentCatalog` to a different namespace, we could move this as well.

I spot checked as many places as I could to remove unnecessary `using Microsoft.ML.Runtime.EntryPoints` statements.

Fix #861

### CODE REVIEW

File: src/Microsoft.ML.Core/Data/ServerChannel.cs [hunk 1]
```diff
@@ -141,7 +141,7 @@ public void Acknowledge(Action<Bundle> toDo)
         /// Entry point factory for creating <see cref="IServer"/> instances.
         /// </summary>
         [TlcModule.ComponentKind("Server")]
-        public interface IServerFactory : IArgsComponent, IComponentFactory<IChannel, IServer>
```
Comment (Ivanidzo4ka) [inline src/Microsoft.ML.Core/Data/ServerChannel.cs]: >IArgsComponent [](start = 42, length = 14)

feel free to remove this interface as well. It was temporary introduce for Neural nets entry points almost two years ago, and we no longer need it.

Comment (eerhardt) [reply src/Microsoft.ML.Core/Data/ServerChannel.cs]: I've removed `IArgsComponent` completely.

**Comment by TomFinley** [comment]
This sounds good thank you @eerhardt...

One thing that we will want to consider is your suggestion earlier that we get rid of all these specialized subinterfaces and just make things over `IComponentFactory` directly. (E.g., no `IServerFactory`, though there would be one of the appropriate `IComponentFactory<`... The only thing that seems to practically stand in the way is all these `[TlcModule.ComponentKind(...)]` declarations -- I'm not sure what they're for but I'd wager they have something to do with entry-points. But I wonder if there's another way we could achieve the same effect. Not related to this, just something that came to mind as I looked at it.

**Comment by eerhardt** [comment]
The existing usages of specialized subinterfaces are all in "Entry Points" APIs. My assumption was that we weren't necessarily working on optimizing these APIs.

**Pull request merged**